pandas/core/indexes/base.py

```diff
@@ -3599,6 +3599,10 @@ def get_loc(self, key, method=None, tolerance=None):
                 return self._engine.get_loc(casted_key)
             except KeyError as err:
                 raise KeyError(key) from err
+            except TypeError:
+                # If we have a listlike key, raise InvalidIndexError instead
+                self._check_indexing_error(key)
+                raise
```

Comment:
should be from err?

Reply:
nope.  if self._check_indexing_error doesn't raise, then we're just re-raising the TypeError.  can flesh out the comment to that extent

Reply:
fleshed out comment + green